hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -804,6 +758,62 @@ SqlResult execute(CreateTypePlan plan) {
+            final PartitioningStrategy<?> strategy = ((MapProxyImpl) hazelcastInstance.getMap(mapName))
+                    .getPartitionStrategy();
+
+            // We only support Default and Attribute strategies, even if one of the maps uses non-Default/Attribute
+            // strategy, we should abort the process and clear list of already populated partitions so that partition
+            // pruning doesn't get activated at all for this query.
+            if (strategy != null
+                    && !(strategy instanceof DefaultPartitioningStrategy)
+                    && !(strategy instanceof AttributePartitioningStrategy)) {
+                allVariantsValid = false;
+                break;
+            }
+
+            // ordering of attributes matters for partitioning (1,2) produces different partition than (2,1).
+            final List<String> orderedKeyAttributes = new ArrayList<>();
+            if (strategy instanceof AttributePartitioningStrategy) {
+                final var attributeStrategy = (AttributePartitioningStrategy) strategy;
+                orderedKeyAttributes.addAll(asList(attributeStrategy.getPartitioningAttributes()));
+            } else {
+                orderedKeyAttributes.add(DEFAULT_UNIQUE_KEY);
+            }
+
+            for (final Map<String, Expression<?>> perMapCandidate : perMapCandidates) {
+                Object[] partitionKeyComponents = new Object[orderedKeyAttributes.size()];
+                for (int i = 0; i < orderedKeyAttributes.size(); i++) {
+                    final String attribute = orderedKeyAttributes.get(i);
+                    if (!perMapCandidate.containsKey(attribute)) {
+                        // Shouldn't happen, defensive check in case Opt logic breaks and produces variants
+                        // that do not contain all the required partitioning attributes.
+                        throw new HazelcastException("Partition Pruning candidate"
+                                + " does not contain mandatory attribute: " + attribute);
+                    }
+                    partitionKeyComponents[i] = perMapCandidate.get(attribute).eval(null, evalContext);
+                }
+
+                final Partition partition = hazelcastInstance.getPartitionService().getPartition(
+                        PartitioningStrategyUtil.constructAttributeBasedKey(partitionKeyComponents)
+                );
+                if (partition == null) {
+                    // Can happen if the cluster is mid-repartitioning/migration, in this case we revert to
+                    // non-pruning logic. Alternative scenario is if the produced partitioning key somehow invalid.
+                    allVariantsValid = false;
+                    break;
+                }
+                partitions.add(partition.getPartitionId());
+            }
+        }
+        return Tuple2.tuple2(partitions, allVariantsValid);
```

Comment:
can't we just return something empty when `!allVariantsValid`?

Reply:
Refactored  78d4a04336c63b2a30eefc3aafba35ce24c282df